addressmanager/src/stores/not_banned_address_store.rs

```diff
@@ -0,0 +1,102 @@
+use database::{
+    db::DB,
+    errors::{StoreError, StoreResult},
+    prelude::{CachedDbAccess, DirectDbWriter},
+};
+use std::net::Ipv6Addr;
+use std::{error::Error, fmt::Display, sync::Arc};
+
+const STORE_PREFIX_CONNECTION_FAILED_COUNT: &[u8] = b"not-banned-addresses-connection-failed-count";
+
+pub trait NotBannedAddressesStoreReader {
+    fn get(&self, address: Ipv6Addr, port: u16) -> Result<u64, StoreError>;
+}
+
+pub trait NotBannedAddressesStore: NotBannedAddressesStoreReader {
+    fn set(&mut self, ip: Ipv6Addr, port: u16, connection_failed_count: u64) -> StoreResult<()>;
+    fn remove(&mut self, ip: Ipv6Addr, port: u16) -> StoreResult<()>;
+}
+
+const IPV6_LEN: usize = 16;
+const PORT_LEN: usize = 2;
+pub const ADDRESS_KEY_SIZE: usize = IPV6_LEN + PORT_LEN;
+
+#[derive(Eq, Hash, PartialEq, Debug, Copy, Clone)]
+struct AddressKey([u8; ADDRESS_KEY_SIZE]);
+
+impl AsRef<[u8]> for AddressKey {
+    fn as_ref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl Display for AddressKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let ip_port: (Ipv6Addr, u16) = (*self).into();
+        write!(f, "{}:{}", ip_port.0, ip_port.1)
+    }
+}
+
```

Comment:
Maybe add a non-urgent TODO to generalize this key serialization scheme (looks very much like utxo index key)